components/navbar.tsx

```diff
@@ -1,18 +1,30 @@
-import React from "react";
+"use client";  // Mark this component as a Client Component
+
+import React, { useEffect, useState } from "react";
 import { ModeToggle } from "./shared/ToggleBg";
 import { GitHubLogoIcon } from "@radix-ui/react-icons";
 import { IoMdGitNetwork } from "react-icons/io";
 import Link from "next/link";
 import { Button } from "./ui/button";
 import AuthButton from "./AuthButton";
 import Hamburger from "./Hamburger";
-import { supabaseServer } from "@/utils/supabase/server";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { usePathname } from "next/navigation";  // Import the usePathname hook
```

Comment:
Remove this

Reply:
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";

only this one or both

Reply:
Both.

Reply:
Is it there any way to find out current path without using 
### import { usePathname } from "next/navigation";  this

If it is can you tell me?
i try this one but it did not work
### const currentPath = typeof window !== "undefined" ? window.location.pathname : "";
@dhairyathedev